catroid/src/main/java/org/catrobat/catroid/ui/ProjectActivity.java

```diff
@@ -335,6 +340,34 @@ public void addSpriteObjectFromUri(final Uri uri, final String extension,
 				.show(getSupportFragmentManager(), NewSpriteDialogFragment.Companion.getTAG());
 	}
 
+	public void addEmptySpriteObject() {
+		final Scene currentScene = ProjectManager.getInstance().getCurrentlyEditedScene();
+
+		View view = View.inflate(this, R.layout.dialog_new_empty_sprite, null);
+
+		String lookDataName =
+				new UniqueNameProvider().getUniqueNameInNameables(getString(R.string.default_sprite_name), currentScene.getSpriteList());
+
+		TextInputDialog.Builder builder = new TextInputDialog.Builder(this);
+
+		builder.setHint(getString(R.string.sprite_name_label))
```

Comment:
In order to reduce code duplication I suggest to reuse the `NewSpriteDialogFragment` here. For example add an additional flag to the class indicating if it is an empty sprite and then set the visibility of the `visuallyPlaceSwitch` to GONE as well as change the text from the `placevisuallyTextView` accordingly. This way there is also no need for creating a separate xml for the dialog.